zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryControllerTest.java

```diff
@@ -0,0 +1,393 @@
+        .search(
+            new UserTaskQuery.Builder()
+                .sort(
+                    new UserTaskSort.Builder().creationDate().desc().completionDate().asc().build())
+                .build());
+  }
+
+  @Test
+  void shouldInvalidateUserTasksSearchQueryWithBadSortOrder() {
+    // given
+    final var request =
+        """
+        {
+            "sort": [
+                {
+                    "field": "creationTime",
+                    "order": "dsc"
+                }
+            ]
+        }""";
+    final var expectedResponse =
+        String.format(
+            """
+        {
+          "type": "about:blank",
+          "title": "Bad Request",
+          "status": 400,
+          "detail": "Unknown sortOrder: dsc",
+          "instance": "%s"
+        }""",
+            USER_TASKS_SEARCH_URL);
+    // when / then
+    webClient
+        .post()
+        .uri(USER_TASKS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON)
+        .expectBody()
+        .json(expectedResponse);
+
+    verify(userTaskServices, never()).search(any(UserTaskQuery.class));
+  }
+
+  @Test
+  void shouldInvalidateUserTasksSearchQueryWithBadSortField() {
+    // given
+    final var request =
+        """
+        {
+            "sort": [
+                {
+                    "field": "unknownField",
+                    "order": "asc"
+                }
+            ]
+        }""";
+    final var expectedResponse =
+        String.format(
+            """
+        {
+          "type": "about:blank",
+          "title": "Bad Request",
+          "status": 400,
+          "detail": "Unknown sortBy: unknownField",
+          "instance": "%s"
+        }""",
+            USER_TASKS_SEARCH_URL);
+    // when / then
+    webClient
+        .post()
+        .uri(USER_TASKS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON)
+        .expectBody()
+        .json(expectedResponse);
+
+    verify(userTaskServices, never()).search(any(UserTaskQuery.class));
+  }
+
+  @Test
+  void shouldInvalidateUserTasksSearchQueryWithMissingSortField() {
+    // given
+    final var request =
+        """
+        {
+            "sort": [
+                {
+                    "order": "asc"
+                }
+            ]
+        }""";
+    final var expectedResponse =
+        String.format(
+            """
+        {
+          "type": "about:blank",
+          "title": "Bad Request",
+          "status": 400,
+          "detail": "Sort field must not be null",
```

Comment:
❌ Heads-up: This will need to change after you rebased your PR since we aligned the problem response handling.
```suggestion
          "type": "about:blank",
          "title": "INVALID_ARGUMENT",
          "status": 400,
          "detail": "Sort field must not be null.",
```